scripts/data_catalog.py

```
@@ -0,0 +1,161 @@
+        try:
+            df = pd.read_csv(file_path, nrows=5)
+            info['rows'] = len(pd.read_csv(file_path))
+            info['columns'] = len(df.columns)
+            
+            if 'datetime' in df.columns or 'timestamp' in df.columns:
+                date_col = 'datetime' if 'datetime' in df.columns else 'timestamp'
+                df_full = pd.read_csv(file_path, usecols=[date_col], parse_dates=[date_col])
+                info['data_start'] = df_full[date_col].min()
+                info['data_end'] = df_full[date_col].max()
+        except Exception:
+            pass
```

Ruff (0.12.2)

`try`-`except`-`pass` detected, consider logging the exception

(S110)